app/controllers/salt_controller.rb

```diff
@@ -0,0 +1,31 @@
+require "velum/salt"
+# HealthController holds methods for checking the readiness of the application.
+class SaltController < ApplicationController
+  skip_before_action :redirect_to_setup
+
+  def update
+    minions = assigned_with_status
+    minions.each do |minion|
+      if minion.update_status == Minion.statuses[:update_needed]
```

Comment:
As a general rule, avoid fat controller methods. In this case, I'd move this logic into an instance method of the `Minion` model. This method would be called in the `assigned_with_status` method when we loop through all the minions (we don't want to loop twice if we can merge both operations).